File: js/euler29.js
function euler29(){
    var seen = new Set();
    for (let a = 2; a <= 100; a++){
        for (let b = 2; b <= 100; b++){
            seen.add(Math.pow(a, b));
        }
    }
    return seen.size;
}

console.log(euler29());

File: go/euler29.go
package main
import (
    "fmt"
    "math/big"
)

func main() {
    seen := make(map[string]bool)
    for a := 2; a <= 100; a++ {
        for b := 2; b <= 100; b++ {
            base := big.NewInt(int64(a))
            exp := big.NewInt(int64(b))
            pow := new(big.Int).Exp(base, exp, nil)
            seen[pow.String()] = true
        }
    }
    fmt.Println(len(seen))
}
